Add `no-arguments-for-html-elements` rule (#927)
ember-template-lint/ember-template-lint: docs/rule/no-arguments-for-html-elements.md

## no-arguments-for-html-elements

Angle bracket components use the `@argument` style and that's why users are more likely to start making this mistake.

For case like:

```hbs
<option @value="1" />
```

we have untrackable error like

```
Uncaught (in promise) TypeError: func is not a function
    at Compilers.compile
    at compile
    at LazyCompiler.add
    at CompilableBlock.compile
    at Object.evaluate
    at AppendOpcodes.evaluate
    at LowLevelVM.evaluateSyscall
    at LowLevelVM.evaluateInner
    at LowLevelVM.evaluateOuter
    at VM.next
```

> This is creepy and untracable (for end-users).

Reviewer: ```suggestion
This is untraceable for end-users.
```

> Using this linting rule we can guess (is it component or not) and emit error if argument assigned to lowercased element without same block name and without dots in notation.

Reviewer: ```suggestion
This linting rule determines if it is a component or not and emits an error if argument is assigned to a lower-case element without the same block name and without dots in notation.
```

> This rule also emit error if we trying to specify "path-like" string as attribute argument.

Reviewer: ```suggestion
This rule will also emit an error for these other conditions: 

- trying to specify "path-like" string as attribute argument
- trying to use an HTML element with block params

```

> Also, error will be thrown if we trying to use HTML element with block-params.

Reviewer: ```suggestion
```

Added this line to the bullet points above

### Examples

This rule **forbids** the following:

```hbs
<picture @name="Cat" >
```

```hbs
<img @src="1" >`
```

```hbs
<img src="this.picture" >`
```

```hbs
<img src="@img" >`
```

```hbs
<div as |blockName|>
    {{blockName}}
</div>
```

This rule **allows** the following:

```hbs
<Picture @name="Cat" />
```

```hbs
<Img @src="1" />
```

```hbs
<img src={{this.picture}} >`
```

```hbs
<img src={{@img}} >`
```

```hbs
<MyComponent as |blockName|>
    {{blockName}}
</MyComponent>
```

```hbs
{{#let (component "bar" as |foo|}}
  <foo @name="1" />
{{/let}}
```

```hbs
<foo.some.name @name="1"/>
<@foo @name="2" />
<@foo.bar @name="2" />
```

```hbs
<MyComponent>

>   <:slot @name="Header"></:slot>

Reviewer: is this legit valid code?

Author: yes, this is "slots" (already available in glimmer-vm)

Reviewer 2: :O Is there documentation on how to use or examples of this? Or is this not recommended to use in production apps yet?

</MyComponent>

> ```

Reviewer: You're missing some of the template.md sections here- having a references section is important at the very least.

Author: @MelSumner could you mention some references? It's removed because I can't figure out matched refs.

> ## no-arguments-for-html-elements

Reviewer 3: Can you mention how "angle bracket components" use the `@argument` style and that's why users are more likely to start making this mistake?

There's also a rule documentation template FYI: https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/_TEMPLATE_.md
